lib/nfg_onboarder/onboarding_controller_helper.rb

```diff
@@ -161,6 +161,8 @@ def on_before_show
       def on_valid_step
         onboarding_session.current_high_level_step = controller_name
         onboarding_session.current_step = next_step
+        onboarding_session.owner = get_event_target
```

Comment:
@pathakh This won't work because during the course of an onboarder we may be creating multiple objects, i.e. the form.model may be different than the actual onboarder. We need to check first whether the onboarding_session owner has already been set, and whether the form_target responds to :onboarding_sessions. So I would suggest this line should be something like:
```
onboarding_session.owner = get_form_target if onboarding_session.owner.nil? && get_form_target.&.onboarding_sessions&.is_a?(ActiveRecord::Associations::CollectionProxy)
```

Reply:
@hoenth  oh ok, i didn't know form.model would be different.  i thought form would be different but it would point to the same model.  sorry.